__tests__/app.test.js

```diff
@@ -135,47 +137,149 @@ describe("GET /api/articles", () => {
-                expect(comment).toMatchObject({
-                    body: expect.any(String),
-                    comment_id: expect.any(Number),
-                    votes: expect.any(Number),
-                    created_at: expect.any(String),
-                    author: expect.any(String),
-                    article_id: expect.any(Number),
-                })
-            });
-            expect(body.comments).toBeSortedBy("created_at", {descending: true});
+describe("GET /api/articles/:article_id/comments", () => {
+  test("200: Responds with all comments for an article", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.comments).toHaveLength(11);
+        body.comments.forEach((comment) => {
+          expect(comment).toMatchObject({
+            body: expect.any(String),
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            article_id: expect.any(Number),
+          });
         });
-    });
-
-    test("404: article ID not found",()=>{
-        return request(app)
-        .get("/api/articles/9999/comments")
-        .expect(404)
-        .then(({body})=>{
-            expect(body.msg).toBe("404 - No comments found for article ID of 9999")
-        })
-    })
-
-    test("400: bad request if banana entered!",()=>{
-        return request(app)
-        .get("/api/articles/banana/comments")
-        .expect(400)
-        .then(({body})=>{
-            expect(body.msg).toBe("400 - Bad request")
-        })
-    })
+        expect(body.comments).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+
+  test("404: article ID not found", () => {
+    return request(app)
+      .get("/api/articles/9999/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("404 - No comments found for article ID of 9999");
+      });
+  });
+
+  test("400: bad request if banana entered!", () => {
+    return request(app)
+      .get("/api/articles/banana/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("400 - Bad request");
+      });
+  });
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: Responds with 201 code and the posted comment", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "AHHHHHHHH test banana grapefruit AHHHH!!!",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then((res) => {
+        expect(res.body.comment).toEqual(
+          expect.objectContaining({
+            body: "AHHHHHHHH test banana grapefruit AHHHH!!!",
+            votes: 0,
+            author: "rogersop",
+            article_id: 1,
+            created_at: expect.any(String),
+          })
+        );
```

Comment:
Really good expect statement, but this is missing the comment_id which is very important and something we should expect to receive.